extension/src/popup/views/MnemonicPhrase/index.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Redirect } from "react-router-dom";
+import { useTranslation } from "react-i18next";
+import { Button, Text, Icon } from "@stellar/design-system";
+
+import { APPLICATION_STATE } from "@shared/constants/applicationState";
+
+import { ROUTES } from "popup/constants/routes";
+import { Onboarding, OnboardingModal } from "popup/components/Onboarding";
+import { ConfirmMnemonicPhrase } from "popup/components/mnemonicPhrase/ConfirmMnemonicPhrase";
+import { DisplayMnemonicPhrase } from "popup/components/mnemonicPhrase/DisplayMnemonicPhrase";
+import {
+  applicationStateSelector,
+  confirmMnemonicPhrase,
+} from "popup/ducks/accountServices";
+import { View } from "popup/basics/layout/View";
+
+import "./styles.scss";
+
+interface MnemonicPhraseProps {
+  mnemonicPhrase: string;
+}
+
+export const MnemonicPhrase = ({
+  mnemonicPhrase = "",
+}: MnemonicPhraseProps) => {
+  const { t } = useTranslation();
+  const applicationState = useSelector(applicationStateSelector);
+  const [isConfirmed, setIsConfirmed] = useState(false);
+  const [isModalShowing, setIsModalShowing] = useState(true);
+  const dispatch = useDispatch();
+
+  const handleSkip = () => {
+    // confirm the mnemonic phrase for the user
+
+    dispatch(confirmMnemonicPhrase(mnemonicPhrase));
+  };
+
+  if (applicationState === APPLICATION_STATE.MNEMONIC_PHRASE_CONFIRMED) {
+    return <Redirect to={ROUTES.mnemonicPhraseConfirmed} />;
+  }
+
+  if (mnemonicPhrase) {
+    if (isModalShowing) {
```

Comment:
compare this file to the deleted `MnemonicPhrase.tsx` file above. It's essentially the same while adding everything in this `isModalShowing` conditional